include/zcbor_debug.h

```diff
@@ -62,6 +62,43 @@ static void zcbor_print_compare_strings_diff(const uint8_t *str1, const uint8_t
 	}
 }
 
+__attribute__((used))
+static const char *zcbor_error_str(int error)
+{
+	#define ZCBOR_ERR_CASE(err) case err: { \
+			static const char err_str[] = #err; \
+			return err_str; \
+		}
+	switch(error) {
+		ZCBOR_ERR_CASE(ZCBOR_SUCCESS)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_NO_BACKUP_MEM)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_NO_BACKUP_ACTIVE)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_LOW_ELEM_COUNT)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_HIGH_ELEM_COUNT)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_INT_SIZE)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_FLOAT_SIZE)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_ADDITIONAL_INVAL)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_NO_PAYLOAD)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_PAYLOAD_NOT_CONSUMED)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_WRONG_TYPE)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_WRONG_VALUE)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_WRONG_RANGE)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_ITERATIONS)
+		ZCBOR_ERR_CASE(ZCBOR_ERR_ASSERTION)
+	}
+	#undef ZCBOR_ERR_CASE
+
+	static const char err_unknown[] = "ZCBOR_ERR_UNKNOWN";
```

Comment:
Yes it would, and it would be equivalent. I'd like to keep it this way, though, because it's slightly less verbose, and to avoid a situation where there is no return statement at the end of the function.

Reply:
Sure :+1: